monday/graphqlclient/client.py

```diff
@@ -1,41 +1,56 @@
-import requests
 import json
 
+import requests
+
+from monday.exceptions import MondayQueryError
+
 
 class GraphQLClient:
     def __init__(self, endpoint):
         self.endpoint = endpoint
         self.token = None
-        self.headername = None
+        self.token_header_name = None
+        self.headers = {}
 
     def execute(self, query, variables=None):
         return self._send(query, variables)
 
-    def inject_token(self, token, headername='Authorization'):
+    def inject_token(self, token, token_header_name='Authorization'):
         self.token = token
-        self.headername = headername
+        self.token_header_name = token_header_name
+
+    def inject_headers(self, headers):
+        self.headers = headers
 
     def _send(self, query, variables):
         payload = {'query': query}
-        headers = {}
+        headers = self.headers.copy()
         files = None
-
+        
         if self.token is not None:
-            headers[self.headername] = '{}'.format(self.token)
+            headers[self.token_header_name] = '{}'.format(self.token)
 
         if variables is None:
-            headers['Content-Type'] = 'application/json'
+            headers.setdefault('Content-Type', 'application/json')
+
             payload = json.dumps({'query': query}).encode('utf-8')
+        
         elif variables.get('file', None) is not None:
-            headers['content'] = 'multipart/form-data'
+            headers.setdefault('content', 'multipart/form-data')
+                
             files = [
                 ('variables[file]', (variables['file'], open(variables['file'], 'rb')))
             ]
 
         try:
             response = requests.request("POST", self.endpoint, headers=headers, data=payload, files=files)
             response.raise_for_status()
+            self._catch_error(response)
```

Comment:
this is not really "catching" errors - it's more validating and throwing an exception if there are errors, so we should probably call it `_throw_on_error(response)` or something

Reply:
also, maybe?

```python
json_response = response.json()
self._raise_for_error(json_response)
return json_response
```

```python
def _raise_for_error(self, data)
    if "errors" in data:
        raise MondayQueryError(data["errors"])
```

Reply:
`_throw_on_error` it is.

Reply:
Also, tell me how you feel about the `MondayQueryError` changes.  I picked a direction and went with it, but I could see more than a few other ways to do it.